### ISSUE TITLE
Need documentation

### ISSUE THREAD

**Issue by escallin** [docusign/DocuSign-.NET-Client#13]
Hey, the library is great, but we need documentation.  The Nuget package does not install any xml documentation for the classes.

Is there documentation anywhere for this library?

**Comment by InbarGazit** [comment]
@escallin , you are correct, we do not yet have documentation. I'm sorry for that, we'll try to get some documentation in the future. In the interim - do you have any questions that I can help you with? We do have some sample code that shows some examples of how to use the library and there are more than 30 unit tests as well that exercise the code, those can be useful as well. I do acknowledge the need for formal documentation, we haven't had a chance to assemble that yet. Let me know if I can assist in any way.

**Comment by escallin** [comment]
Hello Inbar.  Thank you for the quick reply.  I noticed that a lot of the source code IS xml commented.  If you could include all the xml comments as a separate XML Documentation file so that intellisense would work, that would be a good start.

I am trying to accomplish a specific task that you don’t have an example for.  Let me write up a scenario and I will send it to you.  Should I ask it in the “issues” board here at github, or is there a better place?

Thanks,
Eric

**Comment by InbarGazit** [comment]
@escallin , I see, you can get the source here and then would have intelisense, but you want to use the nuget and it's not coming through, right? we would definitely get this fixed soon. I need to figure out how to do that and would get back to you.

As for the scenario - sure, feel free to open an issue here so it's publicly shared with others who might have the same need.

Thanks for your patience!

**Comment by escallin** [comment]
Yes, that’s correct.  The nuget package only downloads the compiled dll.  

I was able to compile the source code from github and output the XML Documentation file.  Once the file is place with the dll from NuGet, the Intellisense will work.  I think you should still include it in the NuGet package so that it updates.

I did notice that the source code is not as well documented as I thought.  Most of the documentation is generic like "Get or Sets the xxxx property".  Really need to now what the property or method is used for.

Thanks,
Eric

**Comment by InbarGazit** [comment]
Thanks Eric, I appreciate the feedback. We would do our best to improve documentation in the future. Please let me know if there's something I can help you with in the interim.

**Comment by Socrathustra** [comment]
Has there been any progress on this? It would be _very_ useful if NuGet installed IntelliSense xml files along with the library. It is quite a pain to have to have the documentation open in one window with Visual Studio in another. People with only one monitor I'm sure are wearing out their ALT and TAB keys.

I did some quick digging, and I think [this](https://docs.microsoft.com/en-us/dotnet/standard/serialization/xml-schema-definition-tool-xsd-exe) is what  you want.

**Comment by LarryKlugerDS** [comment]
Hi Everybody,
First, I do apologize for our (DocuSign's) very slow responses to issues raised in the repo. We've reorganized to improve matters.

We share your desire to document the SDK itself. Since each SDK method has a one-to-one correspondence to an API method, the [API reference documentation](https://developers.docusign.com/esign-rest-api/reference) can often be used.

**Comment by LarryKlugerDS** [comment]
I've entered internal enhancement request DEVDOCS-1148

**Comment by smendoza88** [comment]
Any update on this?
